RefrigeratorExe/RefrigeratorExe/Program.cs

```diff
@@ -0,0 +1,297 @@
+                Console.WriteLine("Press 1 to print out Item in Refrigerator");
+                Console.WriteLine("Press 2 to print the amout of place left in Refrigerator");
+                Console.WriteLine("Press 3 to add item to Refrigeratoe");
+                Console.WriteLine("Press 4 to take out item our of Refrigerator");
+                Console.WriteLine("Press 5 to clean Refigerator");
+                Console.WriteLine("Press 6 What do you want to eat");
+                Console.WriteLine("Press 7 to sort item according to expiry date");
+                Console.WriteLine("Press 8 to sort Shelfs acccording to space");
+                Console.WriteLine("Press 9 to sort refrigerators according to space");
+                Console.WriteLine("Press 10 to get ready for shoping");
+                Console.WriteLine("Prss 100 to exit");
+                numberPress = InputNumber();
+                switch (numberPress)
+                {                   
+                    case 1:
+                        Console.WriteLine(refrigerator.ToString());
+                        break;
+                    case 2:
+                        Console.WriteLine(refrigerator.SpaceInRefrigerator());
+                        break;
+                    case 3:
+                        GetItemFromUser(refrigerator);                      
+                        break;
+                    case 4:
+                        RemoveItemFromRefrigerator(refrigerator);
+                        break;
+                    case 5:
+                        CleanRefrigerator(refrigerator);
+                        break;
+                    case 6:
+                        WhatDoYouWantToEat(refrigerator);
+                        break;
+                    case 7:
+                        CallSortItem(refrigerator);
+                        break;
+                    case 8:
+                        CallSortShelfs(refrigerator);
+                        break;
+                    case 9:
+                        CallSortRefrigerator(refrigerators);
+                        break;
+                    case 10:
+                        ReadyForShopping(refrigerator);
+                        break;
+                    case 100:
+                        flag=false;
+                        break;
+                    default:
+                        break;
+                }
+            }         
+
+        }
+        public static int InputNumber()
+        {
+
+            bool isParse = false;
+            int inputNumber = 0;
+            Console.WriteLine("Please enter one of the folwing numbers");
+            isParse = int.TryParse(Console.ReadLine(), out inputNumber);
+            if (isParse && ((inputNumber > 0) && (inputNumber <= 10) || inputNumber == 100))
+            {
+                return inputNumber;
+            }
+            Console.WriteLine("Invalid number");
+            return InputNumber();
```

Comment:
you used recursion, which is ok, but we have to consider the stack depth in case the program runs for long time and some bad script tries to send it the same wrong value continuously. This is why when it comes to user interface it is preferred to use while(true), like this:

while (true)
{
    Console.WriteLine("Please enter one of the following numbers");
    var isParse = int.TryParse(Console.ReadLine(), out var inputNumber);
    if (isParse && inputNumber is > 0 and <= 10 or 100)
    {
        return inputNumber;
    }

    Console.WriteLine("Invalid number");
}